### PR TITLE
Improve value formatting API, add time zone support

### LINKED ISSUE #80: Add library of tick formatters
There are two issues with the current tick formatting API: #67, #79.

To fix these problems I propose we expose several tick formatting utilities in the vis library, which can handle common cases like formatting times or formatting quantities of bytes.

These utilities could be composed with user specific functions to fix #67.

It would also let us generate ticks more intelligently when the baked-in tick formatting utilities are used, since we can take into account the formatting when generating ticks. This would fix #79.

### PR DESCRIPTION
**Pull request by chnn**
timezones

Suppose a user has a basic Giraffe config:

```
import {Config} from '@influxdata/giraffe'

// ...

const config: Config = {
  table,
  layers: [{
    type: 'line',
    x: 'time',
    y: 'cpuPercentage'
  }]
}
```

Previously, they could control how values from their `table` were displayed using the `valueFormatters` option:

```
import {Config} from '@influxdata/giraffe'

// ...

const config: Config = {
  table,
  valueFormatters: {
    cpuPercentage: x => `${x}%`
  },
  layers: [{
    type: 'line',
    x: 'time',
    y: 'cpuPercentage'
  }]
}
```

However, there were some issues with this:

- It was difficult to make small changes to Giraffe's default formatting of date time values (#67)
- We would generate ugly axis tick marks when a user used custom value formatting (#79)

This PR solves the first issue and enables a solution for the second, while also introducing support for modifying the time zone of a plot. It does this by introducing public utilities for creating formatter functions:

```
import {timeFormatter, Config} from '@influxdata/giraffe'

// ...

const config: Config = {
  table,
  valueFormatters: {
    time: timeFormatter({timeZone: 'America/Chicago'}),
    cpuPercentage: x => `${x}%`
  },
  layers: [{
    type: 'line',
    x: 'time',
    y: 'cpuPercentage'
  }]
}
```

The formatting functions created by utilities like `timeFormatter` are plain functions, but they have also been tagged with a `_GIRAFFE_FORMATTER_TYPE` property. Internally we can recognize such functions and treat them specially when generating ticks. A user can still pass a custom function for format arbitrary columns in their table.

Besides for `timeFormatter`, a `binaryPrefixFormatter` utility and `siPrefixUtility` utility are also introduced. In the future, I think we could add more such utilities (for example, for time durations or imperial distances).

I also added some snapshot tests to catch any regressions in time zone support or binary prefix formatting.

Closes #80

### CODE REVIEW

File: src/utils/formatters.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+export const TIME_FORMATTER_TYPE: 'TIME' = 'TIME'
+
+interface TimeFormatter extends Formatter {
+  (x: number, options?: {domainWidth?: number}): string
+
+  _GIRAFFE_FORMATTER_TYPE: typeof TIME_FORMATTER_TYPE
+}
+
+interface TimeFormatterFactoryOptions {
```
Comment (chnn) [inline src/utils/formatters.ts]: We are using the browser [`Intl`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat) API for zero-dependency time zone and locale support.

File: src/utils/getTicks.ts [hunk 2]
```diff
@@ -1,37 +1,26 @@
...
-    orientation === 'vertical' ? charMetrics.height : charMetrics.width
+  switch (formatter._GIRAFFE_FORMATTER_TYPE) {
+    case 'TIME':
+      return getTimeTicks(domain, rangeLength, numTicks)
 
-  const numTicks = getNumTicks(sampleTick, length, charLength)
-
-  if (columnType === 'time') {
-    return getTimeTicks(domain, length, numTicks)
+    default:
```
Comment (chnn) [inline src/utils/getTicks.ts]: We can solve https://github.com/influxdata/giraffe/issues/79 here by adding another case statement.

File: src/utils/formatters.test.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    // 24 hour time when using UTC
+    expect(utcFormatter(d)).toEqual('1/1/2019, 00:00 UTC')
+
+    // Default (locale specific) when not UTC
+    expect(nonUTCFormatter(d)).toEqual('12/31/2018, 4:00 PM PST')
+  })
+})
+
+describe('siPrefixFormatter', () => {
```
Comment (ebb-tide) [inline src/utils/formatters.test.ts]: Above and beyond! :)

File: src/components/Axes.tsx [hunk 4]
```diff
@@ -58,6 +62,7 @@ export const drawAxes = ({
 
   const context = canvas.getContext('2d')
   const xAxisY = height - margins.bottom
+  const xDomainWidth = xDomain[1] - xDomain[0]
```
Comment (ebb-tide) [inline src/components/Axes.tsx]: nice! just checking there's no chance that xDomain[1] < xDomain[0]?

Comment (chnn) [reply src/components/Axes.tsx]: hmm good point, I think that might be able to happen. I'll update this

**Pull request merged**